src/main/java/hlf/java/rest/client/service/impl/EventPublishServiceImpl.java

```diff
@@ -82,18 +87,51 @@ public void onFailure(Throwable ex) {
   }
 
   @Override
-  public boolean publishChaincodeEvents(
+  public void publishChaincodeEvents(
       String payload,
       String chaincodeName,
       String fabricTxId,
       String eventName,
       String channelName,
       String messageKey) {
-    boolean status = true;
 
+    Optional<FabricProperties.ChaincodeDetails> optionalChaincodeDetails =
+        fabricProperties.getEvents().getChaincodeDetails().stream()
+            .filter(
+                chaincodeDetail ->
+                    chaincodeDetail.getChannelName().equals(channelName)
+                        && chaincodeDetail.getChaincodeId().equals(chaincodeName))
+            .findAny();
+
+    if (!optionalChaincodeDetails.isPresent()
+        || CollectionUtils.isEmpty(optionalChaincodeDetails.get().getListenerTopics())) {
+      sendMessage(
+          kafkaProperties.getEventListeners().get(0).getTopic(),
+          payload,
+          chaincodeName,
+          fabricTxId,
+          eventName,
+          channelName,
+          messageKey);
+      return;
```

Comment:
Why should we publish the chaincode details are not present in the Events config?

Reply:
corrected